conftest.py

```diff
@@ -73,3 +84,174 @@ def undo(self):
 @pytest.fixture(autouse=True)
 def default_patches(request):
     return DefaultPatches(request)
+
+
+@pytest.fixture(scope="session")
+def requires_hardlinks():
+    if not are_hardlinks_supported():
+        pytest.skip("hardlinks not supported")
+
+
+@pytest.fixture(autouse=True)
+def set_env_variables(monkeypatch):
+    os.environ["BORG_CHECK_I_KNOW_WHAT_I_AM_DOING"] = "YES"
+    os.environ["BORG_DELETE_I_KNOW_WHAT_I_AM_DOING"] = "YES"
+    os.environ["BORG_PASSPHRASE"] = "waytooeasyonlyfortests"
+    os.environ["BORG_SELFTEST"] = "disabled"
+
+
+class ArchiverSetup:
+    RK_ENCRYPTION = "--encryption=repokey-aes-ocb"
+    KF_ENCRYPTION = "--encryption=keyfile-chacha20-poly1305"
+    PURE_PYTHON_MSGPACK_WARNING = "Using a pure-python msgpack! This will result in lower performance."
+
+    src_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), "src", "borg", "archiver"))
+
+    EXE: str = None  # python source based
+    FORK_DEFAULT = False
+    prefix = ""
+
+    def __init__(self):
+        self.archiver = None
+        self.tmpdir = str
+        self.repository_path = str
+        self.repository_location = str
+        self.input_path = str
+        self.output_path = str
+        self.keys_path = str
+        self.cache_path = str
+        self.exclude_file_path = str
+        self.patterns_file_path = str
+        self._old_wd = str
+
+
+@pytest.fixture(autouse=True)
+def archiver_setup(tmp_path):
```

Comment:
names of things are all temporary, will refine later on.

Reply:
Right. If a fixture gives back `archiver`, it can be called like that.😜

Reply:
Absolutely 👍 

Lot's of renaming and refinement to do. Tracer Bullet style for now :)